### ISSUE TITLE
[BUG] Incorrect permissions for delete action on BOM items 

### ISSUE THREAD

**Issue by eeintech** [inventree/InvenTree#2409]
**Describe the bug**

Cannot delete a selection of BOM items even with Parts>Change permission enabled.

**Steps to Reproduce**

1. Navigate to part of assembly type
2. Load the Bill of Materials tab
3. Click on the edit BOM button
4. Select a number of items
5. Click on the delete button for the selection (not individual)
6. Click "Accept", nothing happens

**Expected behavior**

All selected BOM items should be deleted from the BOM.

**Deployment Method**
- [ ] Docker
- [X] Bare Metal

**Version Information**
# Version Information:
InvenTree-Version: 0.5.4
Django Version: 3.2.5
Commit Hash: 66a08fe91
Commit Date: 2021-11-03
Database: postgresql
Debug-Mode: False
Deployed using Docker: False

**Comment by SchrodingersGat** [comment]
@eeintech i've just tried to replicate on `0.5.4` without success.. Can you try again? Are you seeing any error messages in the javascript console?

**Comment by eeintech** [comment]
Oops I found out why... the permission for this user profile are:

![image](https://user-images.githubusercontent.com/4020546/144623907-b7d49561-0180-4e20-b4fe-a134d8d98b7a.png)

And the BoM items can only be deleted if Parts-Delete permission is on... I think BoM items should be able to be deleted if Parts-Change permission is on (it's a _change_ on a part). I'll create a PR for that.

**Comment by eeintech** [comment]
Actually @SchrodingersGat now that BomItems are handled through API calls, I am not sure how permissions can be altered...

**Comment by matmair** [comment]
@eeintech you can just override the permission-check on the api view if it the method is 'delete'

**Comment by SchrodingersGat** [comment]
@eeintech happy to close this one, then?

**Comment by eeintech** [comment]
@SchrodingersGat Not really, this is a real issue still. Either the delete button shouldn't be shown or the permission corrected. I was intending to do the later today but didn't have time to dig more. I can look into @matmair comment next week

**Comment by SchrodingersGat** [comment]
Hiding the button would be a pretty straight forward approach, and probably should be done independent of any decision made on permissions.

**Comment by eeintech** [comment]
@SchrodingersGat Is the `wontfix` label for `0.6.0` or only for `0.5.5`?

**Comment by SchrodingersGat** [comment]
I believe this has been fixed in 0.6.x and given the imminent release of 0.6.0 I don't think there will be an 0.5.5

**Comment by eeintech** [comment]
@SchrodingersGat Just checked but it is not fixed in `0.6.x`, see below.

## Group permission
![image](https://user-images.githubusercontent.com/4020546/152186079-a1519c81-313c-4340-aee1-ae362fb574ed.png)

## Attempt to delete all BOM items
![image](https://user-images.githubusercontent.com/4020546/152186297-f73f5d99-c8c5-4bed-bfeb-3721d9486129.png)

![image](https://user-images.githubusercontent.com/4020546/152186467-5ce4c65a-c335-4ec1-a1fb-0e5876983fb8.png)

## Attempt to delete single BOM item
![image](https://user-images.githubusercontent.com/4020546/152186544-83cce93d-6f95-4082-89e1-f9a085fd0909.png)

![image](https://user-images.githubusercontent.com/4020546/152186596-a1fbaed9-3ad7-4060-9ce2-e4d9b0fa4e3b.png)

**Comment by eeintech** [comment]
I started to look into this and it is a bit more complex than I originally thought as `Roles` are the entry point for permissions and the per-model permissions abstracted from `Roles`, example:
- Role `Part` is set to Add+Change
- Role `Part` controls permissions for models `PartParameter`, `BomItem`, etc
- A permission check for a model (eg. `PartParameter`) will solve for the role it is attached to (eg. `Part`)

In summary, if user tries to delete either `PartParameter` or `BomItem` from a part, user will get denied permission.
Unless I'm mistaken, **there is no mechanism** to bypass the role permission as of now.
So if one added the permission to delete `BomItem` (which I started working on), request will still get denied because parent role (eg. Part) has no `delete` permission.

My thought would be to enable a parent-children relationship for permissions, which would be different than the `RULESET_MODELS` mapping.
In this new lookup table, a list of children models could be associated to a parent role. If the parent role has the `change` permission, all the children models action permissions (eg. `add`, `change` and `delete`) will be automatically enabled.
This lookup table should be checked upon permission request (eg. `RuleSet.check_table_permission` should be updated to take it into account).

Going back to the example, This will allow a user to delete a part parameter or a BOM item whenever this user is in a group with `part.change` permission.
